core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -151,11 +164,49 @@ oppia.directive('searchBar', [function() {
             }
 
             _updateSelectionDetails('languageCodes');
+
+            if (Boolean(urlService.getUrlParams().q)) {
+              _parseURL();
+            }
             _onSearchQueryChangeExec();
 
             _searchBarFullyLoaded = true;
           }
         );
+
+        var _parseURL = function() {
```

Comment:
FYI we are moving away entirely from leading underscores in vars, so I suggest not following the existing convention here.

Also, how about updateSearchFieldsBasedOnUrlQuery()? I think that's more descriptive than parseUrl -- it's hard to tell what the latter does just from the name.